### PR TITLE
Add active steps detection

### PR DESCRIPTION
**Pull request by gregorylegarec**
Thanks @m4dz to have a review. (ping also @CPatchane and @misstick)

This PR add a way to detect if an onboarding step is active, i.e. it it has to appear in the onboarding flow.

To handle that, it adds an `isActive(user)` method to Onboarding Step class, which specifies if the step has to be active for a given user (See documentation).

This method may be overriden by config steps. See for example infos step.

### Infos step

Also, it add the infos step, only active when given user don't have valid `email` or `timezone`.

### Also added
* Documentation
* Tests

### CODE REVIEW

**Comment by nono** [comment]
@gregorylegarec coffeelint is not happy :wink:

**Comment by gregorylegarec** [comment]
@nono Oops, sorry. It's the second time that it happens to me, I will see how I could add coffeelint to the npm watch task. :)

**Comment by gregorylegarec** [comment]
@nono Should be fixed now.

File: client/doc/onboarding.md [hunk 1]
```diff
@@ -61,6 +61,30 @@ Go directly to the given step and trigger required events. Useful for go back in
 Returns a step by its name.
 
 ### Step
+
+#### isActive(user)
+* `user`: JS Object
+
+Returns true if the step has to be active for the given `user`. Returns `true` by default.
```
Comment (misstick) [inline client/doc/onboarding.md]: I'm not sure that your first sentence his usefull. It seems obvious according to the method name. Furthermore your exemple below speaks for itself.
Precise default value by the way is great 👍

File: client/doc/onboarding.md [hunk 2]
```diff
@@ -61,6 +61,30 @@ Go directly to the given step and trigger required events. Useful for go back in
...
+This method can be overriden by specifying an `isActive` method in constructor parameter.
+
+##### Example
+```javascript
+let step = new Step({
```
Comment (misstick) [inline client/doc/onboarding.md]: Could you precize where should be this code? Where is it called on this project? Is it Application.initialize()?

**Comment by nono** [comment]
@gregorylegarec you can also remove node 0.10 and node 0.12 in `travis.yml`

File: .travis.yml [hunk 3]
```diff
@@ -8,8 +8,6 @@ branches:
   except:
     - demo
 node_js:
-    - 0.10
-    - 0.12
     - 4
     - 5
```
Comment (gregorylegarec) [inline .travis.yml]: Could we make a dedicated PR for this stuff ? We are starting to mix to concerns here :)

File: client/app/lib/onboarding.coffee [hunk 4]
```diff
@@ -46,11 +54,15 @@ module.exports = class Onboarding
...
-            stepModel.onValidated () =>
-                @handleStepSubmitted()
-            return stepModel
+        @steps = steps
+            .reduce (activeSteps, step) =>
+                stepModel = new Step step
+                if stepModel.isActive user
+                    activeSteps.push stepModel
+                    stepModel.onValidated () =>
+                        @handleStepSubmitted()
```
Comment (gregorylegarec) [inline client/app/lib/onboarding.coffee]: Good question, I asked it to myself. Actually if we do this like it, we lose the onboading `this` context, as CoffeeScript does not offer a way to directly bind a function (or maybe I just don't know how to do). Wrapping this code in another function is the only way I found keep this context.

File: .travis.yml (hunk 3 above)
Comment (m4dz) [inline .travis.yml]: Al'right, I agree 👍

File: client/app/lib/onboarding.coffee (hunk 4 above)
Comment (m4dz) [inline client/app/lib/onboarding.coffee]: Hmm… we can declare in https://github.com/cozy/cozy-proxy/pull/307/files#diff-753b070baa8f3302572f4381d40ac8daR80 

```coffee
handleStepSubmitted: =>
    @goToNext()
```

But if I understand code properly, this method can be overriden in Step, so there's a risk to loose the binding, right?

If this, just take it as it :)

Comment (gregorylegarec) [inline client/app/lib/onboarding.coffee]: For the moment, this method is not designed to be overriden. So let use your solution, thanks 👍

**Comment by m4dz** [comment]
Thanks @gregorylegarec !

Just for information, we'll try to not forget to update localization keys in steps markups 😉
